examples/tv_minimization_reconstruction_pdhg.py

```diff
@@ -0,0 +1,190 @@
+# %% [markdown]
+# # Total-variation (TV)-minimization reconstruction of 2D golden angle radial data
+
+# %%
+# define zenodo URL of the example ismrmd data
+zenodo_url = 'https://zenodo.org/records/10854057/files/'
+fname = 'pulseq_radial_2D_402spokes_golden_angle_with_traj.h5'
+# %%
+# Download raw data
+import tempfile
+
+import requests
+
+data_file = tempfile.NamedTemporaryFile(mode='wb', delete=False, suffix='.h5')
+response = requests.get(zenodo_url + fname, timeout=30)
+data_file.write(response.content)
+data_file.flush()
+
+# %% [markdown]
+# ### Image reconstruction
+# Here, we use the Primal Dual Hybrid Gradient (PDHG) algorithm to reconstruct an image from 2D radial k-space data
+# data.
+#
+# Let $y$ denote the k-space data of the image $x_{\mathrm{true}}$ sampled with an acquisition model $A$
+# (Fourier transform, coil sensitivity maps, ...), i.e the forward problem is given as
+#
+# $ y = Ax_{\mathrm{true}} + n, $
+#
+# where $n$ describes complex Gaussian noise. When using TV-minimization as regularization method, an approximation of
+# $x_{\mathrm{true}}$ is obtained by minimizing the following functional $\mathcal{F}$:
+#
+# $ \mathcal{F}(x) = \frac{1}{2}||Ax - y||_2^2 + \lambda \| \nabla x \|_1, \quad \quad \quad (1)$
+#
+# where $\nabla$ is the discretized gradient operator.
+#
+# The minimization of the functional $\mathcal{F}$ is a non-trivial task due to the presence of the operator
+# $\nabla$ in the non-differentiable $\ell_1$-norm. A suitable algorithm to solve the problem is the
+# PDHG-algorithm [Chambolle \& Pock, JMIV 2011].
+#
+# PDHG is a method for solving problems of the form
+#
+# $ \min_x f(K(x)) + g(x)  \quad \quad \quad (2)$
+#
+# where $f$ and $g$ denote proper, convex, lower-semicontinous functionals and $K$ denotes a linear operator.
+#
+# PDHG essentially consists of three steps, which read as
+#
+# $z_{k+1} = \mathrm{prox}_{\sigma f^{\ast}}(z_k + \sigma K \bar{x}_k)$
+#
+# $x_{k+1} = \mathrm{prox}_{\tau g}(x_k - \tau K^H z_{k+1})$
+#
+# $\bar{x}_{k+1} = x_{k+1} + \theta(x_{k+1} - x_k)$,
+#
+# where $\mathrm{prox}$ denotes the proximal operator and $f^{\ast}$ denotes the convex conjugate of the
+# functional $f$, $\theta\in [0,1]$ and step sizes $\sigma, \tau$ such that $\sigma \tau < 1/L^2$, where
+# $L=\|K\|_2$ is the operator norm of the operator $K$.
+#
+# The first step is to recast problem (1) into the general form of (2) and then to apply the steps above
+# in an iterative fashion.
+#
+# A possible and intuitive (but unfortunately not efficient) identification is the following
+#
+# $f(x) = \lambda \| x\|_1,$
```

Comment:
I would remove this.

It goes beyond the scope of an example ways how not to do it.

Reply:
I would like to keep it... it's just a few lines and compared to the space it takes, it's highly instructional :)

The first time I wanted to use pdhg for TV, I remember being quite puzzled by why one would not use the first identification.

Reply:
Can we maybe move it down into its own section?

I would like to reduce the wall of text before the first code.

Also, for the identification we use, could you conaidee interleaving the explanation with the code a bit more?

I think this could make it more educational